jenkins/agent-ctl.sh

```diff
@@ -304,40 +307,40 @@ gcloud compute ssh "${INSTANCE}" << INSTANTIATE_DONE
 set -o verbose
 set -o errexit
 
+sudo apt-get -y update
+sudo apt-get -y install \
+  apt-transport-https \
+  ca-certificates \
+  curl \
+  gnupg2 \
+  software-properties-common \
+  python-openssl python-pyasn1 python-ndg-httpsclient \
+  build-essential \
+  tmpreaper \
+  jq
+
 # Install docker
-which docker || curl -sSL https://get.docker.com/ | sh
+curl -fsSL https://download.docker.com/linux/debian/gpg | sudo apt-key add -
+sudo apt-key fingerprint 0EBFCD88 | grep '9DC8 5822 9FC7 DD38 854A  E2D8 8D81 803C 0EBF CD88'
+sudo add-apt-repository -y "deb [arch=amd64] https://download.docker.com/linux/debian \$(lsb_release -cs) stable"
+sudo apt-get -y update
+sudo apt-get -y install docker-ce
```

Comment:
https://github.com/kubernetes/test-infra/issues/3549 happened because of this, though a workaround is getting merged into kubernetes now...